vendor/sigs.k8s.io/kustomize/api/internal/git/cloner.go

```diff
@@ -22,15 +22,11 @@ func ClonerUsingGitExec(repoSpec *RepoSpec) error {
 	if err = r.run("init"); err != nil {
 		return err
 	}
-	if err = r.run(
-		"remote", "add", "origin", repoSpec.CloneSpec()); err != nil {
-		return err
-	}
 	ref := "HEAD"
 	if repoSpec.Ref != "" {
 		ref = repoSpec.Ref
 	}
-	if err = r.run("fetch", "--depth=1", "origin", ref); err != nil {
+	if err = r.run("fetch", "--depth=1", repoSpec.CloneSpec(), ref); err != nil {
```

Comment:
previously, adding the clonespec as an origin was unambiguous. inlining it here, will git always interpret this arg as a remote, or is there any value this could hold that git would misinterpret as something else?

Reply:
@KnVerey Could you answer this one?

Reply:
To the best of my knowledge, this is always the "[repository](https://git-scm.com/docs/git-fetch#Documentation/git-fetch.txt-ltrepositorygt)" argument, which can either be the repository URL directly, or a [remote name](https://git-scm.com/docs/git-fetch#REMOTES) that will be used to fetch the repository URL from a configuration file. Since the lines above create a clean configuration in which the CloneSpec is the remote URL (with no further configuration of the remote), I think this is necessarily equivalent?

As for the content of CloneSpec, this is the code that controls it: https://github.com/kubernetes-sigs/kustomize/blob/master/api/internal/git/repospec.go. It is user-provided, but goes through quite a bit of parsing code to validate that it seems like a git URL (and to do things like extract the sub-repo Kustomization path from the user-provided string).

Reply:
ok, thanks